### PR TITLE
Allow parsing long GISDBASE paths

### LINKED ISSUE #1373: [Bug] Cannot build under deep directory structures
**Describe the bug**

Trying to build GRASS 7.8.5, I get many build errors like this:

```
make
if [ "/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/bin/test.raster3d.lib" != "" ] ; then GISRC=/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/demolocation/.grassrc78 GISBASE=/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu PATH="/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/bin:/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/bin:/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/scripts:$PATH" PYTHONPATH="/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/etc/python:/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/gui/wxpython:$PYTHONPATH" LD_LIBRARY_PATH="/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/bin:/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/bin:/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/scripts:/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/lib:/root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/lib:" LC_ALL=C LANG=C LANGUAGE=C /root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu/bin/test.raster3d.lib --html-description < /dev/null | grep -v '</body>\|</html>' > test.raster3d.lib.tmp.html ; fi
access: No such file or directory                                                                                                                 
ERROR: LOCATION                                                                                                                                   
       </root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin
/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-li/demolocation>                                                                                 
       not available                                                                                                                              
```

Note the `ERROR: LOCATION` path is clipped before the `/demolocation` suffix: `dist.x86_64_pc-li`, lacks the `nux-gnu` suffix.

Contents of `/root/.cache/bazel/_bazel_root/7b7747ec045ae6
06eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linu
x-gnu/demolocation/.grassrc78`:
```
GISDBASE: /root/.cache/bazel/_bazel_root/7b7747ec045ae606eb720a1222f56098/sandbox/processwrapper-sandbox/64/execroot/wasm/bazel-out/k8-fastbuild/bin/lib/grass/grass_lib.build_tmpdir/dist.x86_64-pc-linux-gnu
LOCATION_NAME: demolocation
MAPSET: PERMANENT
GUI: text
```

If I modify the `GISDBASE` in the `.grassrc78` file during build, and copy it to a location with a shorter path, the build passes.

**To Reproduce**

Similar to #1327:

```
git clone git@github.com:attilaolah/wasm
cd wasm
docker build -t wasm docker
docker run -it -v "${PWD}:/build" wasm
```

Then once inside the container, build like so:

```
apt install python3 python3-six
bazel build //lib/grass --sandbox_debug
```

It will download the same compiler and everything, build dependencies first and then attempt to build grass itself.

**Expected behavior**
It should build.

**Screenshots**

**System description (please complete the following information):**

- Operating System: Ubuntu 20.10 from the Docker hub
- GRASS GIS version 7.8.5, haven't tried other versions so far.

### PR DESCRIPTION
**Pull request by attilaolah**
Use a larger buffer when parsing the `.gisrc` file. Use `GPATH_MAX` to fit whatever would fit in the path value, plus the length of the key, plus 2 for newline character and null byte.

Fixes #1373.

### CODE REVIEW

File: lib/gis/env.c [hunk 1]
```diff
@@ -115,8 +115,8 @@ void G__read_gisrc_env(void)
 }
 
 static void parse_env(FILE *fd, int loc)
-{    
-    char buf[200];
+{
+    char buf[GPATH_MAX + strlen("GISDBASE: \0\n")];
```
Comment (marisn) [inline lib/gis/env.c:119]: Just hardcode the length and add a comment explaining it.

**Review by marisn — APPROVED** [review]

File: lib/gis/env.c (hunk 1 above)
Comment (attilaolah) [reply lib/gis/env.c:119]: You mean hardcode `strlen("GISDBASE: \0\n")`, or the whole thing, GPATH_MAX + strlen("GISDBASE: \0\n"`?

I'm assuming the former.

Comment (attilaolah) [reply lib/gis/env.c]: Not sure if you meant a more detailed comment; how about this one?

Comment (nilason) [reply lib/gis/env.c]: something like `GPATH_MAX + 10`

general note: strlen only counts up till null character, `\0\n` is superfluous

Comment (attilaolah) [reply lib/gis/env.c]: Ah, right. I added `+ 12` to account for the `\n` and the `\0` which are I think also returned in the buffer, so `GPATH_MAX + 12`.

Comment (attilaolah) [reply lib/gis/env.c]: (But maybe my comment is incorrect then…)

Comment (nilason) [reply lib/gis/env.c]: Please put the comment on a separate line above declaration of `char buf[GPATH_MAX + 12]` (with C style comment), eg.:

```
/* account for long lines up to GPATH_MAX  */
```
and why not round up the 12 chars to 16.

Comment (attilaolah) [reply lib/gis/env.c]: Done. Sorry I forgot about the C-style comments. Not that often I touch pure C files these days.

**Review by nilason — APPROVED** [review]

**Pull request merged**